asteroid/engine/gan_system.py

```diff
@@ -0,0 +1,161 @@
+from collections import OrderedDict
+import soundfile as sf
+import pytorch_lightning as pl
+import torch
+import os
+
+
+class GanSystem(pl.LightningModule):
+
+    def __init__(self, discriminator, generator, opt_d, opt_g,
+                 scheduler_d, scheduler_g, discriminator_loss, generator_loss,
+                 validation_loss, train_loader, val_loader=None, conf=None):
```

Comment:
I don't think the schedulers should be positional arguments (meaning mandatory), they should be keyword args with None as default. 
Is there a good default value for validation loss or is it mandatory? I think we can make it a kwarg as well.

Reply:
I guess stoi is the best choice.